### PR TITLE
Implement nanmin function in keras.ops

### PR DESCRIPTION
**Pull request by shashaka**
Adds keras.ops.nanmin, which computes the minimum of elements along a specified axis while ignoring NaN values in the input tensor.
The operation follows NumPy semantics for axis and keepdims and is supported across NumPy, TensorFlow, PyTorch, and JAX backends. Not supported on OpenVINO.

### CODE REVIEW

File: keras/src/backend/torch/numpy.py [hunk 1]
```diff
@@ -1272,6 +1272,31 @@ def moveaxis(x, source, destination):
...
+
+    out = torch.amin(x_clean, dim=axis, keepdim=keepdims)
+
+    if axis is not None:
+        all_nan = torch.isnan(x).all(dim=axis, keepdim=keepdims)
+        out = torch.where(all_nan, torch.nan, out)
+    elif torch.isnan(x).all():
+        out = torch.tensor(float("nan"), dtype=x.dtype)
+
+    return out
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py]: 1. torch.amin does NOT return (values, indices) — it always returns a tensor. Only torch.min does that.
2. torch.nanmin does NOT exist in PyTorch (including 2.9.1).

File: keras/src/backend/tensorflow/numpy.py [hunk 2]
```diff
@@ -2125,6 +2125,35 @@ def moveaxis(x, source, destination):
...
+    out = tf.reduce_min(x_clean, axis=axis, keepdims=keepdims)
+
+    if axis is not None:
+        all_nan = tf.reduce_all(tf.math.is_nan(x), axis=axis, keepdims=keepdims)
+        out = tf.where(all_nan, tf.constant(float("nan"), dtype=x.dtype), out)
+    else:
+        if tf.reduce_all(tf.math.is_nan(x)):
+            out = tf.constant(float("nan"), dtype=x.dtype)
+
+    return out
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: TensorFlow does not have tf.math.reduce_nanmin, so the implementation cannot be simplified that way.

File: keras/src/backend/tensorflow/numpy.py [hunk 3]
```diff
@@ -2125,6 +2125,35 @@ def moveaxis(x, source, destination):
...
+
+    if not x.dtype.is_floating:
+        dtype = standardize_dtype(x.dtype)
+        if dtype == "bool":
+            return cast(
+                tf.reduce_min(
+                    cast(x, config.floatx()), axis=axis, keepdims=keepdims
+                ),
+                dtype,
+            )
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: I think it would be easier and more efficient to just do:

```python
return tf.reduce_all(x, axis=axis, keepdims=keepdims)
```

File: keras/src/backend/tensorflow/numpy.py [hunk 4]
```diff
@@ -2125,6 +2125,35 @@ def moveaxis(x, source, destination):
...
+    inf = tf.constant(float("inf"), dtype=x.dtype)
+    x_clean = tf.where(tf.math.is_nan(x), inf, x)
+
+    out = tf.reduce_min(x_clean, axis=axis, keepdims=keepdims)
+
+    if axis is not None:
+        all_nan = tf.reduce_all(tf.math.is_nan(x), axis=axis, keepdims=keepdims)
+        out = tf.where(all_nan, tf.constant(float("nan"), dtype=x.dtype), out)
+    else:
+        if tf.reduce_all(tf.math.is_nan(x)):
```
Comment (hertschuh) [inline keras/src/backend/tensorflow/numpy.py]: You cannot do an `if` on a tensor result, it won't be compilable (we turn off Autograph).

You'll need to use a `tf.where` or `tf.cond` here.

File: keras/src/backend/torch/numpy.py [hunk 5]
```diff
@@ -1272,6 +1272,31 @@ def moveaxis(x, source, destination):
...
+    x = convert_to_tensor(x)
+    if not torch.is_floating_point(x):
+        return torch.amin(x, dim=axis, keepdim=keepdims)
+
+    if axis == () or axis == []:
+        return x
+
+    x_clean = torch.where(
+        torch.isnan(x),
+        torch.full((), float("inf"), device=x.device, dtype=x.dtype),
```
Comment (hertschuh) [inline keras/src/backend/torch/numpy.py]: You don't need `torch.full` here, you can just have a scalar tensor for `float("inf")`.

File: keras/src/backend/torch/numpy.py [hunk 6]
```diff
@@ -1272,6 +1272,31 @@ def moveaxis(x, source, destination):
...
+        x,
+    )
+
+    out = torch.amin(x_clean, dim=axis, keepdim=keepdims)
+
+    if axis is not None:
+        all_nan = torch.isnan(x).all(dim=axis, keepdim=keepdims)
+        out = torch.where(all_nan, torch.nan, out)
+    elif torch.isnan(x).all():
```
Comment (hertschuh) [inline keras/src/backend/torch/numpy.py]: Similar comment here about `if` with a tensor. While it should typically work on torch, I don't know what happens if you try to use a compiler.

File: keras/src/ops/numpy_test.py [hunk 7]
```diff
@@ -5473,6 +5503,45 @@ def test_slogdet(self):
         self.assertEqual(out[0].shape, ())
         self.assertEqual(out[1].shape, (2, 4))
 
+    def test_nanmin(self):
+        x = np.array([[1.0, np.nan, 3.0], [np.nan, 2.0, 1.0]])
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py]: Can you change it to 
```
x = np.array([[1.0, np.nan, 3.0], [np.nan, 2.0, np.inf]])
```
to exercise the infinite case too?

File: keras/src/backend/tensorflow/numpy.py (hunk 3 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: Thanks, I’ve updated the code.

File: keras/src/backend/tensorflow/numpy.py (hunk 4 above)
Comment (shashaka) [reply keras/src/backend/tensorflow/numpy.py]: I’ve updated the code.

File: keras/src/backend/torch/numpy.py (hunk 5 above)
Comment (shashaka) [reply keras/src/backend/torch/numpy.py]: Thanks, I’ve updated it to use a scalar tensor for float("inf") instead of torch.full.

File: keras/src/backend/torch/numpy.py (hunk 6 above)
Comment (shashaka) [reply keras/src/backend/torch/numpy.py]: I've updated.

File: keras/src/ops/numpy_test.py (hunk 7 above)
Comment (shashaka) [reply keras/src/ops/numpy_test.py]: I've applied.

File: keras/src/backend/torch/numpy.py [hunk 8]
```diff
@@ -1272,6 +1272,24 @@ def moveaxis(x, source, destination):
     return torch.moveaxis(x, source=source, destination=destination)
 
 
+def nanmin(x, axis=None, keepdims=False):
+    x = convert_to_tensor(x)
+    if not torch.is_floating_point(x):
+        return torch.amin(x, dim=axis, keepdim=keepdims)
+
+    if axis == () or axis == []:
+        return x
```
[the remark this answers is not in the document]
Comment (shashaka) [reply keras/src/backend/torch/numpy.py:1281]: To align its behavior with NumPy, I added this code.

File: keras/src/ops/numpy_test.py [hunk 9]
```diff
@@ -5473,9 +5503,47 @@ def test_slogdet(self):
...
+            ]
+        )
+        self.assertAllClose(knp.nanmin(x_3d), np.nanmin(x_3d))
+        self.assertAllClose(
+            knp.nanmin(x_3d, axis=(1, 2)),
+            np.nanmin(x_3d, axis=(1, 2)),
+        )
+
+    def test_nansum(self):
+        x = np.array([[1.0, np.nan, 3.0], [np.nan, 2.0, np.inf]])
```
Comment (hertschuh) [inline keras/src/ops/numpy_test.py]: Oh, you changed `test_nansum` instead of `nanmin`.

File: keras/src/backend/openvino/excluded_concrete_tests.txt [hunk 10]
```diff
@@ -16,6 +16,8 @@ NumpyDtypeTest::test_inner
 NumpyDtypeTest::test_isreal
 NumpyDtypeTest::test_lcm
 NumpyDtypeTest::test_ldexp
+NumpyDtypeTest::test_logaddexp2
```
Comment (hertschuh) [inline keras/src/backend/openvino/excluded_concrete_tests.txt]: Remove

File: keras/src/backend/openvino/excluded_concrete_tests.txt [hunk 11]
```diff
@@ -39,6 +41,7 @@ NumpyDtypeTest::test_view
 NumpyDtypeTest::test_vstack
 HistogramTest
 NumpyOneInputOpsCorrectnessTest::test_angle
+NumpyOneInputOpsCorrectnessTest::test_argpartition
```
Comment (hertschuh) [inline keras/src/backend/openvino/excluded_concrete_tests.txt]: Remove

File: keras/src/backend/openvino/excluded_concrete_tests.txt [hunk 12]
```diff
@@ -49,6 +52,8 @@ NumpyOneInputOpsCorrectnessTest::test_diagflat
 NumpyOneInputOpsCorrectnessTest::test_exp2
 NumpyOneInputOpsCorrectnessTest::test_imag
 NumpyOneInputOpsCorrectnessTest::test_isreal
+NumpyOneInputOpsCorrectnessTest::test_logaddexp2
```
Comment (hertschuh) [inline keras/src/backend/openvino/excluded_concrete_tests.txt]: Remove

Comment (shashaka) [reply keras/src/backend/openvino/excluded_concrete_tests.txt]: Removed.

File: keras/src/backend/openvino/excluded_concrete_tests.txt (hunk 11 above)
Comment (shashaka) [reply keras/src/backend/openvino/excluded_concrete_tests.txt]: Removed.

File: keras/src/backend/openvino/excluded_concrete_tests.txt (hunk 10 above)
Comment (shashaka) [reply keras/src/backend/openvino/excluded_concrete_tests.txt]: Removed.

File: keras/src/ops/numpy_test.py (hunk 9 above)
Comment (shashaka) [reply keras/src/ops/numpy_test.py]: Thanks for catching that. I corrected.

**Review by hertschuh — APPROVED** [review]
